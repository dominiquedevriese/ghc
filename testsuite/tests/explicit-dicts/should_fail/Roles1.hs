{-# LANGUAGE RoleAnnotations #-}
{-# LANGUAGE TypeApplications #-}
module Roles1 where

import GHC.Base (Ord(..))

type role Set nominal
data Set a

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

f1 = (undefined :: Ord a => a -> f a -> f a)
     ((reverseOrd @Int :: Ord a))
