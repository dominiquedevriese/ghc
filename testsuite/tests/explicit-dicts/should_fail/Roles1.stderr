
Roles1.hs:13:6: error:
    • Explicit dictionary application not allowed
    • In the expression:
        (undefined :: Ord a => a -> f a -> f a)
          ((reverseOrd @Int :: Ord a))
      In an equation for ‘f1’:
          f1
            = (undefined :: Ord a => a -> f a -> f a)
                ((reverseOrd @Int :: Ord a))
